### ISSUE TITLE
resolvePath incorrect work with empty key, incorrect json path escape

### ISSUE THREAD

**Issue by Feverqwe** [mobxjs/mobx-state-tree#1128]
**_Bug report_**

Hello! I have some problems with resolve and escape json path:

```javascript
import { types, getPath, resolvePath, escapeJsonPath } from "mobx-state-tree";
import { escapePathComponent } from "fast-json-patch";

const Store = types
  .model({
    map: types.map(
      types.model({
        model: types.model({
          value: types.string
        })
      })
    )
  })
  .actions(self => {
    return {
      setMap(key, value) {
        self.map.set(key, { model: { value } });
      }
    };
  });
const store = Store.create();
store.setMap("", "value");
const path = getPath(store.map.get("").model);
console.log("path", path);
try {
  console.log("value", resolvePath(store, path));
} catch (err) {
  console.error("resolvePath error", err);
}

const value = "http://exmaple.com";
console.log("mobx-state-tree escape", escapeJsonPath(value));
console.log("fast-json-patch escape", escapePathComponent(value));
```

**Sandbox link or minimal reproduction code**
https://codesandbox.io/s/pyyvzljno7 (see console)

**Describe the expected behavior**
Resolve path must work
Json path escape must be correct.

**Describe the observed behavior**
Resolve path throw error
Json path escape is incorrect

**Comment by Feverqwe** [comment]
Now resolvePath works different with relative target. It is ok?
3.10.0 https://codesandbox.io/s/6n0y0l5q3z
3.10.1 https://codesandbox.io/s/nnrk9pyo4j

**Comment by xaviergonz** [comment]
true, there's a regression there, reopening

**Comment by xaviergonz** [comment]
should be fixed in 3.10.2
